### PR TITLE
feature/gameplay_feature_action_grant

### PR DESCRIPTION
**Pull request by TheEmidee**
Implemented GASExtGameFeatureAction_Grant

### CODE REVIEW

**Review by Frickerson — APPROVED** [review]

File: Source/GASExtensions/Private/GameFeatures/GASExtGameFeatureAction_Grant.cpp [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+                AddedExtensions.Abilities.Add( ability_handle );
+            }
+        }
+
+        for ( const auto & attributes : abilities_entry.GrantedAttributes )
+        {
+            if ( !attributes.AttributeSetType.IsNull() )
+            {
+                TSubclassOf< UAttributeSet > attribute_set_class = attributes.AttributeSetType.LoadSynchronous();
```
Comment (Frickerson) [inline Source/GASExtensions/Private/GameFeatures/GASExtGameFeatureAction_Grant.cpp:225]: auto + cast

File: Source/GASExtensions/Private/GameFeatures/GASExtGameFeatureAction_Grant.cpp [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+                    ability_system_component->AddAttributeSetSubobject( new_set );
+                }
+            }
+        }
+
+        for ( const auto & effect : abilities_entry.GrantedEffects )
+        {
+            if ( !effect.IsNull() )
+            {
+                TSubclassOf< UGameplayEffect > gameplay_effect_class = effect.LoadSynchronous();
```
Comment (Frickerson) [inline Source/GASExtensions/Private/GameFeatures/GASExtGameFeatureAction_Grant.cpp:247]: auto + cast

File: Source/GASExtensions/Private/GameFeatures/GASExtGameFeatureAction_Grant.cpp [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+    }
+
+    if ( make_component_request )
+    {
+        const auto * world = actor->GetWorld();
+        const auto * game_instance = world->GetGameInstance();
+
+        if ( auto * component_manager = UGameInstance::GetSubsystem< UGameFrameworkComponentManager >( game_instance ) )
+        {
+            const TSharedPtr< FComponentRequestHandle > request_handle = component_manager->AddComponentRequest( abilities_entry.ActorClass, component_type );
```
Comment (Frickerson) [inline Source/GASExtensions/Private/GameFeatures/GASExtGameFeatureAction_Grant.cpp]: auto

**Review by Frickerson — CHANGES_REQUESTED** [review]

File: Source/GASExtensions/Private/GameFeatures/GASExtGameFeatureAction_Grant.cpp (hunk 3 above)
Comment (TheEmidee) [reply Source/GASExtensions/Private/GameFeatures/GASExtGameFeatureAction_Grant.cpp]: done

File: Source/GASExtensions/Private/GameFeatures/GASExtGameFeatureAction_Grant.cpp (hunk 2 above)
Comment (TheEmidee) [reply Source/GASExtensions/Private/GameFeatures/GASExtGameFeatureAction_Grant.cpp:247]: not needed there's an implicit conversion

File: Source/GASExtensions/Private/GameFeatures/GASExtGameFeatureAction_Grant.cpp (hunk 1 above)
Comment (TheEmidee) [reply Source/GASExtensions/Private/GameFeatures/GASExtGameFeatureAction_Grant.cpp:225]: not needed there's an implicit conversion (and it's easier to read than a cast IMO)

**Review by Frickerson — APPROVED** [review]

**Pull request merged**